app/Jobs/CreateRockTheVotePostInRogue.php

```diff
@@ -39,35 +39,42 @@ public function __construct($record)
      */
     public function handle(Rogue $rogue)
     {
-    	$shouldProcess = $this->scrubRecord($this->record);
+    	$isTestData = $this->isTestData($this->record);
 
-        if ($shouldProcess) {
+        if (!$isTestData) {
             info('progress_log: Processing: ' . $this->record['Email address']);
 
-            $referralCode = $this->record['Tracking Source'];
-            $referralCodeValues = $this->parseReferralCode($referralCode);
+            $referralCodeValues = $this->parseReferralCode($this->record['Tracking Source']);
+            info('Referral code: ' . implode(', ', $referralCodeValues));
 
-            $user = $this->getOrCreateUser($this->record, $referralCodeValues);
+            $userId = $referralCodeValues['northstar_id'];
+            $campaignId = (int) $referralCodeValues['campaign_id'];
 
-            // @TODO: If we write more functions that are identical across all voter reg imports, pull out into a ImportsVoterReg trait
-            $this->updateNorthstarStatus($user, $this->translateStatus($this->record['Status'], $this->record['Finish with State']));
+            if (!isset($userId) || empty($userId)) {
```

Comment:
This can just be `if (empty($userId))`, since we know the variable will be defined at this point.